tests/unit/test_clidriver.py

```diff
@@ -737,6 +739,32 @@ def test_idempotency_token_is_not_required_in_help_text(self):
             self.driver.main(['servicecatalog', 'create-constraint'])
         self.assertNotIn('--idempotency-token', self.stderr.getvalue())
 
+    @mock.patch('awscli.clidriver.distro.id', return_value='amzn')
+    @mock.patch('awscli.clidriver.distro.major_version', side_effect='1')
+    def test_user_agent_for_linux(self, *args):
+        driver = create_clidriver()
+        user_agent_extra_pattern = re.compile(
+            r'^.+/[a-z1-9_]+\.amzn\.1$'
+        )
+        self.assertIsNotNone(user_agent_extra_pattern.match(
+            driver.session.user_agent_extra))
+
+    def test_user_agent(self, *args):
+        driver = create_clidriver()
+        user_agent_extra_pattern = re.compile(
+            r'^.+/[a-z1-9_]+(\.[a-z]+)?(\.[1-9]+)?$'
+        )
+        self.assertIsNotNone(user_agent_extra_pattern.match(
+            driver.session.user_agent_extra))
+        # check that distro didn't fail
+        self.assertFalse('linux' in driver.session.user_agent_extra)
```

Comment:
We may want to remove this assertion if the distro has `linux` in the name in case the distribution id has `linux`.